src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,130 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_CAPACITY = 16;
+    static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        this.table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if ((float) size / table.length >= LOAD_FACTOR) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(key, value, null);
+        int index = getKeyHashcode(key) != 0
+                ? Math.abs(getKeyHashcode(newNode.key)) % table.length : 0;
+        Node<K, V> node = table[index];
+        if (node != null) {
+            while (true) {
+                if (Objects.equals(node.key, newNode.key)) {
+                    node.value = newNode.value;
+                    return;
+                }
+                if (node.next != null) {
+                    node = node.next;
+                } else {
+                    break;
+                }
+            }
+            node.next = newNode;
+        } else {
+            table[index] = newNode;
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = key != null ? Math.abs(key.hashCode()) % table.length : 0;
+        Node<K, V> node = table[index];
+        if (node != null) {
+            while (true) {
+                if (Objects.equals(key, node.key)) {
+                    return node.value;
+                }
+                if (node.next != null) {
+                    node = node.next;
+                } else {
+                    break;
+                }
+            }
+        }
```

Comment:
```suggestion
        while (node != null) {
            if (Objects.equals(key, node.key)) {
                return node.value;
            }
            if (node.next != null) {
               node = node.next;
            } else {
                break;
            }
       }
```
lets also combine if and while

Reply:
Done